test/fetchAndCache.test.js

```diff
@@ -0,0 +1,74 @@
+import fetchAndCache from '../src/utils/fetchAndCache';
+
+describe('fetchAndCache', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  class Response {
+    constructor(body, opts) {
+      this.body = body;
+      if (opts.headers) {
+        this.headers = {
+          get(key) {
+            return opts.headers[key];
+          },
+        };
+      }
+    }
+
+    clone() {
+      return this;
+    }
+  }
+
+  it('directly fetches if cache is not supported', async () => {
+    global.fetch = jest.fn(() => Promise.resolve('FOO'));
+    const res = await fetchAndCache('http://example.com');
+    expect(global.fetch).toHaveBeenLastCalledWith('http://example.com');
+    expect(res).toBe('FOO');
+  });
+
+  it('returns cached response if available and not stale', async () => {
+    const mockResponse = new Response('FOO', {
+      headers: { Expires: Date.now() + 10000 },
+    });
+    global.caches = {
+      open: async () => ({ match: async () => mockResponse }),
+    };
+    global.fetch = jest.fn(() => Promise.resolve('I SHOULD NOT BE RESOLVED'));
+
+    const res = await fetchAndCache('http://example.com');
+    expect(res).toBe(mockResponse);
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('calls fetch if there is no cached copy', async () => {
```

Comment:
I think the only test scenario left is "we call fetch and we cache the response":
https://codecov.io/gh/mozilla-frontend-infra/firefox-performance-dashboard/pull/233/diff#D3-28

Reply:
Yes. Had that in mind but had to mock too much (including Header), so decided to leave it.

Reply:
Tried to mock them again, but too much mocking and not worth it. If https://github.com/jsdom/jsdom/issues/1724 gets fixed, we won't have to mock (Jest uses JSDOM)